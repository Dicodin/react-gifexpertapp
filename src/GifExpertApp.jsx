import React, { useState } from "react";
import { AddCategory, GifGrid } from "./components";

const GifExpertApp = ({ defaultCategories = [] }) => {

    // const categories = ['Samurai Champloo', 'Samurai X', 'Death Note'];
    // const [categories, setCategories] = useState(['Samurai Champloo']);
    const [categories, setCategories] = useState(defaultCategories);

    return (
        <>
            <h2>GifExpertApp</h2>
            <AddCategory setCategories={setCategories}></AddCategory>
            <hr></hr>

            <ol>
                {
                    // El key es obligatorio y no puede ser el Ã­ndice del arreglo
                    categories.map(category => (
                        <GifGrid
                            key={category}
                            category={category}>
                        </GifGrid>
                    ))
                }
            </ol>
        </>
    );

};

export default GifExpertApp;