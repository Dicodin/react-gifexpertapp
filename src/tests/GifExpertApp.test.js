import { shallow } from "enzyme";
import GifExpertApp from "../GifExpertApp";


describe('Pruebas sobre <GifExpertApp></GifExpertApp>', () => {

    test('debe de mostrar de forma correcta el componente ', () => {

        const wrapper = shallow(<GifExpertApp></GifExpertApp>);
        expect(wrapper).toMatchSnapshot();

    });

    test('debe de mostrar una lista de categorias ', () => {

        const categories = ['Tao', 'Buda'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}></GifExpertApp>);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });

});
