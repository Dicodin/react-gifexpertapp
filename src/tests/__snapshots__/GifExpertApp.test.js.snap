// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <GifExpertApp></GifExpertApp> debe de mostrar de forma correcta el componente  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas sobre <GifExpertApp></GifExpertApp> debe de mostrar una lista de categorias  1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Tao"
      key="Tao"
    />
    <GifGrid
      category="Buda"
      key="Buda"
    />
  </ol>
</Fragment>
`;
